src/app/pages/RoflAppsPage/index.tsx

```diff
@@ -1,88 +1,128 @@
+}) => {
+  const { wantedNamePattern, clearFilters } = useROFLAppFiltering()
 
-const limit = NUMBER_OF_ITEMS_ON_SEPARATE_PAGE
+  const {
+    tablePagination,
+    isLoading,
+    roflApps,
+    limit,
+    hasNoResultsOnSelectedPage,
+    hasNoResultsBecauseOfFilters,
+  } = useRoflApps(network, layer)
+
+  if (hasNoResultsOnSelectedPage) {
+    throw AppErrors.PageDoesNotExist
+  }
+
+  if (hasNoResultsBecauseOfFilters) {
+    return <NoMatchingDataMaybeClearFilters clearFilters={clearFilters} />
+  }
+
+  return (
+    <>
+      {tableView === TableLayout.Horizontal && (
+        <RoflAppsList
+          apps={roflApps}
+          isLoading={isLoading}
+          limit={limit}
+          pagination={tablePagination}
+          highlightedPartOfName={wantedNamePattern}
+        />
+      )}
+
+      {tableView === TableLayout.Vertical && (
+        <VerticalList>
+          {isLoading &&
+            [...Array(limit).keys()].map(key => (
+              <RoflAppDetailsVerticalListView key={key} isLoading={true} app={undefined} />
+            ))}
+          {!isLoading &&
+            roflApps?.map(app => (
+              <RoflAppDetailsVerticalListView
+                key={app.id}
+                app={app}
+                highlightedPartOfName={wantedNamePattern}
+              />
+            ))}
+        </VerticalList>
+      )}
+    </>
+  )
+}
 
 export const RoflAppsPage: FC = () => {
-  const [tableView, setTableView] = useState<TableLayout>(TableLayout.Horizontal)
+  const { tableView, setTableView } = useTableViewMode()
   const { t } = useTranslation()
   const { isMobile } = useScreenSize()
-  const pagination = useSearchParamsPagination('page')
-  const offset = (pagination.selectedPage - 1) * limit
-  const scope = useRequiredScopeParam()
 
-  if (scope.layer !== Layer.sapphire) {
+  const { wantedNameInput, setWantedNameInput, nameError } = useROFLAppFiltering()
+  const { network, layer } = useRequiredScopeParam()
+
+  if (layer !== Layer.sapphire) {
     throw AppErrors.UnsupportedLayer
   }
 
-  useEffect(() => {
-    if (!isMobile) {
-      setTableView(TableLayout.Horizontal)
-    }
-  }, [isMobile, setTableView])
-
-  const roflAppsQuery = useGetRuntimeRoflApps(scope.network, scope.layer, {
-    limit: tableView === TableLayout.Vertical ? offset + limit : limit,
-    offset: tableView === TableLayout.Vertical ? 0 : offset,
-  })
-  const { isLoading, isFetched, data } = roflAppsQuery
-  const roflApps = data?.data.rofl_apps
-
-  if (isFetched && pagination.selectedPage > 1 && !roflApps?.length) {
-    throw AppErrors.PageDoesNotExist
-  }
+  const { pagination, isLoading, hasNoResultsBecauseOfFilters, jumpToSingleResult } = useRoflApps(
+    network,
+    layer,
+  )
+  const searchBar = (
+    <TableSearchBar
+      value={wantedNameInput}
+      onChange={setWantedNameInput}
+      placeholder={t('rofl.searchByNameOrKeyword')}
+      warning={nameError}
+      fullWidth={isMobile}
+      // autoFocus={!isMobile}
```

Comment:
This wasn't breaking matomo e2e tests, they are just flaky. I re-ran the CI on previous commit and it passed now :shrug:

Reply:
Ah ok, in that case I'm re-enabling it.